tools/scrimlet/softnpu-init.sh

```diff
@@ -13,27 +13,31 @@ echo "Using $GATEWAY_IP as gateway ip"
 GATEWAY_MAC=${GATEWAY_MAC:=$(arp "$GATEWAY_IP" | awk -F ' ' '{print $4}')}
 echo "Using $GATEWAY_MAC as gateway mac"
 
-# Sidecar Interface facing "sled"
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" port create 1:0 100G RS
+z_swadm () {
+    pfexec zlogin oxz_switch /opt/oxide/dendrite/bin/swadm $@
+}
 
-# Sidecar Interface facing external network
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" port create 2:0 100G RS
 
-# Configure sidecar local ipv6 addresses
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" addr add 1:0 fe80::aae1:deff:fe01:701c
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" addr add 2:0 fe80::aae1:deff:fe01:701d
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" addr add 1:0 fd00:99::1
-
-# Configure route to oxide rack
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" route add fd00:1122:3344:0101::/64 1:0 fe80::aae1:deff:fe00:1
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" arp add fe80::aae1:deff:fe00:1 a8:e1:de:00:00:01
-
-# Configure default route
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" route add 0.0.0.0/0 2:0 "$GATEWAY_IP"
-./out/softnpu/swadm -h "[fd00:1122:3344:101::2]" arp add "$GATEWAY_IP" "$GATEWAY_MAC"
+# Add front facing port
+z_swadm port create 1:0 100G RS
```

Comment:
Same comment as in the A-to-Z doc, we should move to the link-based scheme at some point. I have https://github.com/oxidecomputer/omicron/pull/2771 open for a different reason that does that. I can merge after this one and do that update, if that's useful.

Reply:
That would be helpful indeed!